add constant labels to gauges and cumulative metrics

paivagustavo opened the pull request:
This PR adds support for constant labels on gauges. 

The proposed syntax is:
```go
r := NewRegistry()

 	f, _ := r.AddFloat64Gauge("TestGaugeWithConstLabel",
		WithLabelKeys("k1"), WithConstLabel("const", "same"), WithConstLabel("const2", "same2"))
```

@rghetia this is my first contribution and as you were the one who created the issue, could you review this?

closes #1112

rghetia commented on metric/registry.go:
```diff
@@ -61,13 +62,21 @@ func WithUnit(unit metricdata.Unit) Options {
 }
 
+// WithConstLabel applies provided constant label.
+func WithConstLabel(labelKey, labelValue string) Options {
```
it might be better to take multiple pairs of constant key/value. Perhaps a map would be better.
```go
func WithConstLabel(constLabel map[LabelKey]LabelValue) Options {
}

paivagustavo replied:
changed.

paivagustavo commented:
@rghetia I've rebased to dev and change the target branch (I may have pick some commits from the master that was not on dev yet, is it a problem?)

While doing this, I noticed that the `canonicalize` was not deterministic and sometimes the `TestGauge` was failing, I have fixed this and made it more reliable while keeping the same purpose.

rghetia commented:
> @rghetia I've rebased to dev and change the target branch (I may have pick some commits from the master that was not on dev yet, is it a problem?)

yes, that will be a problem. you can  do `git rebase --onto dev HEAD~2` since you two commits.

paivagustavo commented:
Done, the PR now contains only the correct commits.

rghetia commented on metric/cumulative_test.go:
```diff
@@ -83,6 +84,62 @@ func TestCumulative(t *testing.T) {
+	e.Inc(5)
+	e, _ = f.GetEntry(metricdata.NewLabelValue("k1v1"))
+	e.Inc(1)
+	e, _ = f.GetEntry(metricdata.NewLabelValue("k1v1"))
```
nit: e is not used again. Please remove it.

paivagustavo replied:
remove :)
